lib/phoenix/live_dashboard/page_builder.ex

```diff
@@ -684,16 +697,64 @@ defmodule Phoenix.LiveDashboard.PageBuilder do
       ...>      }
       ...>    ]
       ...> ]
-      iex> layered_graph(layers: layers, title: "My Graph", hint: "A simple graph")
   """
-  @spec layered_graph(keyword()) :: component()
-  def layered_graph(assigns) do
-    assigns =
-      assigns
-      |> Map.new()
-      |> LayeredGraphComponent.normalize_params()
-
-    {LayeredGraphComponent, assigns}
+  attr :id, :any,
+    required: true,
+    doc: "Because is a stateful `Phoenix.LiveComponent` an unique id is needed."
+
+  attr :title, :string, default: nil, doc: "The title of the component."
+
+  attr :hint, :string, default: nil, doc: "A textual hint to show close to the title."
+
+  attr :layers, :list,
```

Comment:
Same as usage. Perhaps it works better as slots?

Reply:
I changed to using slots. I'm not pretty sure if it is a good change. It is a pity that slots within slots cannot be used; it would be a much better solution in this case.

On the other hand, I don't like the `:data` field for the node, it is polymorphic. Wouldn't it be a better solution to have just four fields?:
- id: we can force it to be a string
- children (with default to `[]`)
- label (with default to id)
- data, it may be anything, it is just to store info.

And then, the functions (background, format_label, format_detail) would receive the whole map with the four fields.

In fact, I tried to apply the previous change, but it was more complicated than I expected. And also we would have to change the broadway dashboard.

Reply:
Oh, I see, it is a nested list. Ok, I think we should revert it to lists as it was then. Sorry for pointing to the wrong direction.